bspump/kafka/sink.py

```diff
@@ -115,6 +127,30 @@ def __init__(self, app, pipeline, connection, id=None, config=None):
 		pipeline.PubSub.subscribe("bspump.pipeline.start!", self._on_start)
 		pipeline.PubSub.subscribe("bspump.pipeline.stop!", self._on_stop)
 
+		# Run watchdog to check the throttling
+		watchdog_thread = threading.Thread(target=self._watchdog)
+		watchdog_thread.daemon = True  # Daemonize the thread to ensure it exits with the main program
+		watchdog_thread.start()
+
+
+	def _watchdog(self):
+		"""
+		Periodically checks the size of the producer
+		queue in order to determine if the buffer size
+		was exceeded and throttling did not work properly.
+		"""
+
+		while True:
+
+			# Check if the configured threshold passed
+			if len(self.Producer) < self.WatchdogWatermark:
+				time.sleep(60)  # Sleep one minute (one cycle)
+				continue
+
+			L.critical("The watermark.watchdog was exceeded in KafkaSink likely b/c throttling of the pipeline was not effective. Stopping the application!")
```

Comment:
Eh, this is definitively not a good approach to solve the issue.
It will only result in restarting environments ... why do we need this?